ipn/ipnauth/ipnauth_windows.go

```diff
@@ -64,7 +64,23 @@ func (t *token) IsAdministrator() (bool, error) {
 		return false, err
 	}
 
-	return t.t.IsMember(baSID)
+	isMember, err := t.t.IsMember(baSID)
+	if err != nil {
+		return false, err
+	}
+	if isMember {
+		return true, nil
+	}
+
+	// Try to obtain a linked token, and if present, check it.
+	// (This should be the elevated token associated with limited UAC accounts.)
+	linkedToken, err := t.t.GetLinkedToken()
+	if err != nil {
+		return false, err
+	}
+	defer linkedToken.Close()
+
+	return linkedToken.IsMember(baSID)
```

Comment:
This seems reasonable; `GetLinkedToken` calls `GetTokenInformation` which returns `TOKEN_LINKED_TOKEN` and thus the linked token. If you wanted to be extra careful, you could probably put all this inside a check that calls `GetTokenInformation(t.t, TokenElevationType, ...)` and then only checks the linked token if the elevation type is `TokenElevationTypeLimited`... though I'm not sure if that matters.

Reply:
Yeah, I thought about that too. Since it's on both our minds, I've added that.